booking.js

```diff
@@ -0,0 +1,256 @@
+/*När användaren klickar på “Book room” ska en ruta öppnas över allt annat, rutans innehåll byts ut i tre steg
+I första steget visas en input där användaren kan välja ett datum, samt en knapp för att gå vidare. När användaren går vidare ska ett anrop göras till API för att se vilka tider som finns lediga det angivna datumet
+
+I andra steget visas inputs för namn och e-post, samt två rullgardinsmenyer. I den första ska användaren kunna välja de tider som fanns tillgängliga, och i den andra hur många deltagare som ska medverka. Alternativen ska motsvara den data som hämtats från API. Slutligen även en knapp för att gå vidare.
+
+När användaren går vidare ska ett tackmeddelande visas, samt möjlighet att stänga ned rutan och återgå till listan över utmaningar
+Bokningen ska kunna öppnas både från startsidan och från den nya sidan, d.v.s. på båda de platser där rum finns. */
+
+// Here im going to do the script for the booking-section
+//Import challenges function from fetchchallenges.js
+import { challenges } from "./fetchChallenges.js";
+/*Import renderchallenges function from fetchchallenges.js*/
+import { renderChallenges } from "./fetchChallenges.js";
+
+//Declaring global variables containing html elements
+const bookingContainer = document.querySelector(".booking-container");
+
+const bookingButtonOne = document.querySelector(".booking-container__button");
+
+const submitBookingButton = document.querySelector(
+  ".booking-container__submit-button"
+);
+
+const challengeTemplate = document.getElementById("card-template");
+
+//Function that fetches data from api
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (response.ok) {
+    return await response.json();
+  } else {
+    throw new Error(
+      `Something went wrong with the request. Error code: ${response.status} ${response.statusText}`
+    );
+  }
+}
+//Declaring empty global variables
+let cardId;
+
+let dateApi;
+//Declaring variable and puts an array of all .sidescroll__btn inside
+/*Fetch Dates api */
+console.log(challenges);
+
+/*challenges.forEach((obj) => console.log(obj.id));*/
+
+challenges.forEach((obj) => {
+  const challengeIds = obj.id.toString();
+  console.log(challengeIds);
+
+  async function fetchDatesApi() {
```

Comment:
FetchDatesApi(), is not used anywhere in the the code its only used to console log the value that it returns, is this correct?